pop-api/src/v0/assets/use_cases/fungibles.rs

```diff
@@ -0,0 +1,372 @@
+/// - allowance
+/// - transfer
+/// - transfer_from
+/// - approve
+/// - increase_allowance
+/// - decrease_allowance
+
+/// Returns the total token supply for a given asset ID.
+///
+/// # Arguments
+/// * `id` - The ID of the asset.
+///
+/// # Returns
+/// The total supply of the token, or an error if the operation fails.
+pub fn total_supply(id: AssetId) -> Result<Balance> {
+	pallets::assets::total_supply(id)
+}
+
+/// Returns the account balance for the specified `owner` for a given asset ID. Returns `0` if
+/// the account is non-existent.
+///
+/// # Arguments
+/// * `id` - The ID of the asset.
+/// * `owner` - The account whose balance is being queried.
+///
+/// # Returns
+/// The balance of the specified account, or an error if the operation fails.
+pub fn balance_of(id: AssetId, owner: AccountId) -> Result<Balance> {
+	pallets::assets::balance_of(id, owner)
+}
+
+/// Returns the amount which `spender` is still allowed to withdraw from `owner` for a given
+/// asset ID. Returns `0` if no allowance has been set.
+///
+/// # Arguments
+/// * `id` - The ID of the asset.
+/// * `owner` - The account that owns the tokens.
+/// * `spender` - The account that is allowed to spend the tokens.
+///
+/// # Returns
+/// The remaining allowance, or an error if the operation fails.
+pub fn allowance(id: AssetId, owner: AccountId, spender: AccountId) -> Result<Balance> {
+	pallets::assets::allowance(id, owner, spender)
+}
+
+/// Transfers `value` amount of tokens from the caller's account to account `to`, with additional
+/// `data` in unspecified format.
+///
+/// # Arguments
+/// * `id` - The ID of the asset.
+/// * `to` - The recipient account.
+/// * `value` - The number of tokens to transfer.
+///
+/// # Returns
+/// Returns `Ok(())` if successful, or an error if the transfer fails.
+pub fn transfer(
+	id: AssetId,
+	to: impl Into<MultiAddress<AccountId, ()>>,
+	value: Balance,
+) -> Result<()> {
+	pallets::assets::transfer(id, to, value)
+}
+
+/// Transfers `value` tokens on behalf of `from` to account `to` with additional `data`
+/// in unspecified format. If `from` is equal to `None`, tokens will be minted to account `to`. If
+/// `to` is equal to `None`, tokens will be burned from account `from`.
+///
+/// # Arguments
+/// * `id` - The ID of the asset.
+/// * `from` - The account from which the tokens are transferred.
+/// * `to` - The recipient account.
+/// * `value` - The number of tokens to transfer.
+///
+/// # Returns
+/// Returns `Ok(())` if successful, or an error if the transfer fails.
+pub fn transfer_from(
+	id: AssetId,
+	from: Option<impl Into<MultiAddress<AccountId, ()>>>,
+	to: Option<impl Into<MultiAddress<AccountId, ()>>>,
+	value: Balance,
+	_data: &[u8],
+) -> Result<()> {
+	match (from, to) {
+		(None, Some(to)) => pallets::assets::mint(id, to, value),
+		(Some(from), None) => pallets::assets::burn(id, from, value),
+		(Some(from), Some(to)) => pallets::assets::transfer_approved(id, from, to, value),
+		_ => Ok(()),
```

Comment:
any source for this?

Reply:
This is removed. Based on the existence of the `Mintable` and `Extendable` trait `transfer_from` will simply transfer approved amount of tokens (i.e. `transfer_approved`)